cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -1481,5 +1481,62 @@ TEST(ExecPlan, SourceEnforcesBatchLimit) {
   }
 }
 
+TEST(ExecPlanExecution, SortAndFetch) {
```

Comment:
Can you add a test where the fetch offset is greater than 0?

Reply:
Updated the test case.